nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/PutSplunkHTTP.java

```diff
@@ -288,4 +232,76 @@ private FlowFile enrichFlowFile(final ProcessSession session, final FlowFile flo
     }
+
+    private static class RequestDetails {
+        private static final String ENDPOINT = "/services/collector/raw";
+
+        private final String endpoint;
+        private final String contentType;
+        private final String charset;
+
+        private RequestDetails(final String endpoint, final String contentType, final String charset) {
+            this.endpoint = endpoint;
+            this.contentType = contentType;
+            this.charset = charset;
+        }
+
+        public String getEndpoint() {
+            return endpoint;
+        }
+
+        public String getContentType() {
+            return contentType;
+        }
+
+        public String getCharset() {
+            return charset;
+        }
+
+        public static RequestDetails getInstance(final ComponentLog logger, final ProcessContext context, final FlowFile flowFile) {
+            final String contentType = (context.getProperty(CONTENT_TYPE).isSet()) ? context.getProperty(CONTENT_TYPE).evaluateAttributeExpressions(flowFile).getValue() : null;
+            final String charset = context.getProperty(CHARSET).evaluateAttributeExpressions(flowFile).getValue();
+            final Map<String, String> queryParameters = new HashMap<>();
+
+            if (context.getProperty(SOURCE_TYPE).isSet()) {
+                queryParameters.put("sourcetype", context.getProperty(SOURCE_TYPE).evaluateAttributeExpressions(flowFile).getValue());
+            }
+
+            if (context.getProperty(SOURCE).isSet()) {
+                queryParameters.put("source", context.getProperty(SOURCE).evaluateAttributeExpressions(flowFile).getValue());
+            }
+
+            if (context.getProperty(HOST).isSet()) {
+                queryParameters.put("host", context.getProperty(HOST).evaluateAttributeExpressions(flowFile).getValue());
+            }
+
+            if (context.getProperty(INDEX).isSet()) {
+                queryParameters.put("index", context.getProperty(INDEX).evaluateAttributeExpressions(flowFile).getValue());
+            }
+
+            return new RequestDetails(getEndpoint(logger, queryParameters), contentType, charset);
+        }
+
+        private static String getEndpoint(final ComponentLog logger, final Map<String, String> queryParameters) {
+            final StringBuilder result = new StringBuilder(ENDPOINT);
+
+            if (!queryParameters.isEmpty()) {
+                final List<String> parameters = new LinkedList<>();
+
+                try {
+                    for (final Map.Entry<String, String> parameter : queryParameters.entrySet()) {
+                        parameters.add(URLEncoder.encode(parameter.getKey(), "UTF-8") + '=' + URLEncoder.encode(parameter.getValue(), "UTF-8"));
+                    }
+                } catch (final UnsupportedEncodingException e) {
+                    logger.error("Could not be initialized because of: {}", new Object[]{e.getMessage()}, e);
```

Comment:
```suggestion
                    logger.error("Could not be initialized because of: {}", e.getMessage(), e);
```

Reply:
Thank you very much for reviewing! I can see your point, if any review comment ends up with a commit, I will include this

Reply:
As I touched the code due to @tpalfy's comments, I realised that this is an instance of `ComponentLogger` not a generic `Logger`. My understanding i. that the suggested parametrisation is not supported by this class, thus I will keep it as it is.